### PR TITLE
Adds possibility to add permissions to getPackageAndUpgradePermissions

### PR DESCRIPTION
**Pull request by welzfabian**

### CODE REVIEW

File: src/main/java/sirius/biz/packages/PackageData.java [hunk 1]
```diff
@@ -32,6 +35,12 @@
     @Part
     private static Packages packages;
 
+    @Parts(AdditionalPackagePermissionsProvider.class)
+    private static PartCollection<AdditionalPackagePermissionsProvider> additionalPackagePermissionsProviders;
+
+    @Transient
+    private final BaseEntity<?> parent;
```
Comment (andyHa) [inline src/main/java/sirius/biz/packages/PackageData.java]: owner?

File: src/main/java/sirius/biz/tenants/TenantData.java [hunk 2]
```diff
@@ -218,7 +218,7 @@
      * Contains the package, upgrades and additional and revoked permissions for this tenant.
      */
     public static final Mapping PACKAGE_DATA = Mapping.named("packageData");
-    private final PackageData packageData = new PackageData();
```
Comment (andyHa) [inline src/main/java/sirius/biz/tenants/TenantData.java]: why not initialize in place?

File: src/main/java/sirius/biz/packages/AdditionalPackagePermissionsProvider.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+public interface AdditionalPackagePermissionsProvider {
+
+    /**
+     * Adds additional roles to the given permissionConsumer, based on the given {@link PackageData} and parent.
+     *
+     * @param packageData        the {@link PackageData} instance this is called from
+     * @param parent             the parent entity of the {@link PackageData}
+     * @param permissionConsumer the consumer for the additonal permissions
+     */
+    void addAdditionalPermissions(PackageData packageData, BaseEntity<?> parent, Consumer<String> permissionConsumer);
```
Comment (andyHa) [inline src/main/java/sirius/biz/packages/AdditionalPackagePermissionsProvider.java]: why not add a getParent / getOwner in packageData?

File: src/main/java/sirius/biz/tenants/TenantData.java (hunk 2 above)
Comment (welzfabian) [reply src/main/java/sirius/biz/tenants/TenantData.java]: because `this` is not the tenant, only TenantData

**Pull request merged**